src/components/MediaViewer/MediaViewer.tsx

```diff
@@ -351,500 +366,615 @@ const MediaViewer = ({ resource }: { resource: CloudinaryResource }) => {
+                  <Button
+                    variant="ghost"
+                    className="text-white"
+                    onClick={() => setDeletion({ state: "confirm" })}
+                  >
+                    <Trash2 className="h-6 w-6" />
+                    <span className="sr-only">Delete</span>
+                  </Button>
+                )}
+                {process.env.NEXT_PUBLIC_PHOTOBOX_MODE === "read-only" && (
+                  <TooltipProvider delayDuration={200}>
+                    <Tooltip>
+                      <TooltipTrigger
+                        className={`${buttonVariants({ variant: "ghost" })} text-zinc-500 hover:text-zinc-500 bg-transparent hover:bg-transparent`}
+                        aria-label="Deleting is disabled"
+                      >
+                        <Trash2 className="h-6 w-6" />
+                        <span className="sr-only">Delete</span>
+                      </TooltipTrigger>
+                      <TooltipContent>
+                        <p>Deleting is Disabled</p>
+                      </TooltipContent>
+                    </Tooltip>
+                  </TooltipProvider>
+                )}
+              </li>
+            )}
+            {isTrash && (
+              <li>
+                <Button
+                  variant="ghost"
+                  className="text-white"
+                  onClick={handleOnRestore}
+                >
+                  <History className="h-6 w-6" />
+                  <span className="sr-only">Restore</span>
+                </Button>
+              </li>
+            )}
+          </ul>
+        </Container>
+
+        {/** Asset viewer */}
+
+        <div className="relative flex justify-center items-center align-center w-full h-full">
+          <CldImage
+            key={`${JSON.stringify(transformations)}-${version}`}
+            className="object-contain"
+            width={resource.width}
+            height={resource.height}
+            src={resource.public_id}
+            alt={`Image ${resource.public_id}`}
+            style={imgStyles}
+            version={version}
+            placeholderStyle="dark"
+            {...transformations}
+          />
+        </div>
       </div>
-    </div>
-  )
+
+      {/** Confirmation dialog to display if there are unsaved edits */}
+
+      <Dialog open={showConfirmDialog}>
+        <DialogContent>
+          <DialogHeader>
+            <DialogTitle>Save Changes?</DialogTitle>
+          </DialogHeader>
+          <p>
+            You have unsaved changes that will be lost. Do you still want to
+            navigate back to the main gallery?
+          </p>
+          <DialogFooter>
+            <Button
+              onClick={() => {
+                router.back();
+              }}
+            >
+              Yes
+            </Button>
+            <Button
+              onClick={() => {
+                setShowConfirmDialog(false);
```

Comment:
I thought it made sense to just close the dialog instead of closing it and saving the media changes immediately, then returning to the main gallery view.

We could do it all in one go, but I thought the UX made more sense to just close the dialog and let them save things like they normally would prior to this PR.

Reply:
funny enough this actually confused me. i had an assumption when using it that the Yes would save, not continue navigating back. will follow up on the main thread

Reply:
Feedback addressed in 9e85431395bfd9897bf0b7fa40301f8785bf4f16